fxtest/printer.go

```diff
@@ -34,3 +37,23 @@ func NewTestPrinter(t TB) fx.Printer {
 func (p *testPrinter) Printf(format string, args ...interface{}) {
 	p.Logf(format, args...)
 }
+
+type testLogger struct {
+	TB
+}
+
+func (t *testLogger) Log(entry fxlog.Entry) {
+
+}
+
+func (t *testLogger) PrintSupply(interface{}) {
+
+}
+
+func (t *testLogger) PrintProvide(interface{}) {
+
+}
+
+func NewLogger(t TB) fxlog.Logger {
+	return &testLogger{t}
```

Comment:
I appreciate the intent but: if fxlog.Logger is private, then this shouldn't be public.

Also, my expectation for a test logger would be that it logs to the testing.TB, which is not the case here.

To make that true, we can export from fxlog, a means of building a `fxlog.Logger` from a `*zap.Logger`. And then we can simply pass a `zaptest.NewLogger(t)` to it.